magda-search-api/src/main/scala/au/csiro/data61/magda/search/elasticsearch/ElasticSearchQueryer.scala

```diff
@@ -80,34 +82,38 @@ class ElasticSearchQueryer(indices: Indices = DefaultIndices)(
     "publisher.acronym")
 
   override def search(
+    jwtToken: Option[String],
     inputQuery: Query,
     start: Long,
     limit: Int,
     requestedFacetSize: Int) = {
     val inputRegionsList = inputQuery.regions.toList
 
+    val publishingStatusQueryFuture = opaQueryer.publishingStateQuery(inputQuery.publishingState, jwtToken)
```

Comment:
👍  I think it's a good idea to separate OPA evaluation result parsing logic &  OPA evaluation result to ES query translation logic.  
But I have to admit that I've done a quick job here --- the ES query is directly translated from OPA response JSON and I actually used Elastic4s' query data structure (in order to save some code to define my own ones) to do some further merge/calculation (e.g. true AND unsolved => unsolved & false AND unsolved => false).
I actually work an improved typescript version parser/evaluator program that will produce some mid result rather than the final SQL query or ES Query.
We probably can translate it back to the scala once it's done

Reply:
OK as long as it's on the roadmap 👍

Reply:
A better parser (and a separate SQL transalator) is available on the PR for issue/2155~